### PR TITLE
Add Java 11 images (latest and debug) / add more tags

### PR DESCRIPTION
**Pull request by chanseokoh**
About #236.

~~This is on top of #283, so #283 should be reviewed first.~~

Debian stretch recently back-ported `openjdk-11`. The Java 11 images get the `openjdk-11-jre-headless` (in parity with the Java 8 images) from the `stretch-backports` snapshot archive.

`gcr.io/distroless/java:latest` and `gcr.io/distroless/java:debug` still point to Java 8.

Two new images:
`gcr.io/distroless/java:11`
`gcr.io/distroless/java:11-debug`

Additional tags:
`gcr.io/distroless/java:8`
`gcr.io/distroless/java:8-debug`
`gcr.io/distroless/java:11`
`gcr.io/distroless/java:11-debug`
`gcr.io/distroless/java/jetty:java8`
`gcr.io/distroless/java/jetty:java8-debug`

### CODE REVIEW

**Comment by chanseokoh** [comment]
Oops, hit the enter key mistakenly. This has to wait until #283 is merged.

**Comment by loosebazooka** [comment]
So this generally matches the format of other java images: `java:<version>-<more_descriptors>`, but should we also include the detailed version numbers? for example `java:8u181-xxx` or `java:11.0.1-xxx` as a secondary tag? I'm not really sure at that point that the different vendors will even share version numbers :\, so we might even have to include vendor?

**Comment by chanseokoh** [comment]
Yeah, we do plan to add more detailed tags along the way. It's just that I don't do everything in this PR.

I think we can switch vendors for existing tags. Of course, we can add tags with vendors too (later).

**Comment by briandealwis** [comment]
Can we put the version and vendor information as image labels?  Putting this information into the tags will get pretty unwieldy.   And there's no choice being offered to consumers: we're not planning to publish multiple vendors, and we're not maintaining parallel streams (e.g., 11.0.1 with security updates now that 11.0.2 is released).

**Comment by briandealwis** [comment]
@chanseokoh were you going to offer jetty with java11 too?

**Comment by chanseokoh** [comment]
Took out #283 (advancing the snapshot), so this PR is standalone and no longer depend on #283.

File: cloudbuild.yaml [hunk 1]
```diff
@@ -31,19 +31,49 @@ steps:
   args: ['tag', 'bazel/java:java8', 'gcr.io/$PROJECT_ID/java:latest']
 
 - name: gcr.io/cloud-builders/bazel
-  args: ['run', '//java:debug']
+  args: ['run', '//java:java8']
+- name: gcr.io/cloud-builders/docker
```
Comment (loosebazooka) [inline cloudbuild.yaml]: I'm not sure how this whole section works, but do we need to rerun the bazel build of the image for each tag of the same build?

looks like we have
```
- name: gcr.io/cloud-buidlers/bazel
  args: ['run', '//java:java8']
```
on line 28 and line 33

Comment (chanseokoh) [reply cloudbuild.yaml]: Ah, good catch. I think we can just tag it multiple times. Will updated it.

Note, it's possible that we will remove this `cloudbuild.yaml` at any point (#286).

File: cloudbuild.yaml [hunk 2]
```diff
@@ -29,21 +29,39 @@ steps:
   args: ['run', '//java:java8']
 - name: gcr.io/cloud-builders/docker
   args: ['tag', 'bazel/java:java8', 'gcr.io/$PROJECT_ID/java:latest']
+- name: gcr.io/cloud-builders/docker
```
[the remark this answers is not in the document]
Comment (ImJasonH) [reply cloudbuild.yaml]: When you `blaze run` a `docker_build` target, it builds the image and puts it in the local Docker daemon tagged as (I believe) `bazel/[path/to]:target`. This looks right to me, and most importantly if running the build works, you've got it correct. `docker tag` will fail if the `src` tag isn't found locally.

[the remark this answers is not in the document]
Comment (loosebazooka) [reply cloudbuild.yaml]: so the docker daemon is shared between containers?

File: java/BUILD [hunk 3]
```diff
@@ -23,12 +23,14 @@ cacerts_java(
...
     ],
     symlinks = {
-        "/usr/bin/java": "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java",
+        "/usr/bin/java": java_executable_path,
     },
     tars = [":cacerts_java"],
-) for mode in [
-    "",
-    ":debug",
+) for (rule_name, jre_deb, java_executable_path) in [
```
Comment (loosebazooka) [inline java/BUILD]: Probably just get some bazel person to look at this, is there a way to share the java_executable_path reference or would that make this section look super wonky?
@nlopezgi wdyt?

Comment (loosebazooka) [reply java/BUILD]: I don't feel super strongly about this. I was concerned about each java path could potentially be shared for each java8, java11 option, but it seems like it's not worth the trouble right now

**Pull request merged**